src/deploy/functions/params.ts

```diff
@@ -267,83 +366,298 @@ export async function resolveParams(
         "Parameter " + param.name + " has default value " + paramDefault + " of wrong type"
       );
     }
-    paramValues[param.name] = await promptParam(param, paramDefault);
+    paramValues[param.name] = await promptParam(param, projectId, paramDefault);
   }
 
   return paramValues;
 }
 
+/**
+ * Handles a SecretParam, either by retrieving its latest value from Cloud Secret Manager if present,
+ * or prompting the user for the value of a new secret.
+ * Always returns a string, since secret values are stored as untyped bytes in CSM.
+ */
+async function handleSecret(secretParam: SecretParam, projectId: string): Promise<string> {
+  const metadata = await getSecretMetadata(projectId, secretParam.name, "latest");
+  if (!metadata.secret) {
+    const secretValue = await promptOnce({
+      name: secretParam.name,
+      type: "password",
+      message: `This secret will be stored in Cloud Secret Manager (https://cloud.google.com/secret-manager/pricing) as ${
+        secretParam.name
+      } and managed by Firebase Hosting (Firebase Hosting Service Agent will be granted Secret Admin role on this secret).\nEnter a value for ${
```

Comment:
I don't think this is true - especially the Hosting SA part. We always use end user credentials in Firebase CLI don't we?

Reply:
Yup.